schemas/session-replay/_common-shape-wireframe-schema.json

```diff
@@ -0,0 +1,69 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "_common-shape-wireframe-schema.json",
+  "title": "CommonShapeWireframe",
+  "type": "object",
+  "description": "Schema of common properties for ShapeWireframe events type and all its sub - types.",
+  "allOf": [
+    {
+      "$ref": "_common-wireframe-schema.json"
+    },
+    {
+      "required": [
+        "opacity",
+        "cornerRadius"
```

Comment:
Why making these two `required`? IMO:
- both have obvious default value for the player (`opacity: 1`, `cornerRadius: 0`);
- in practise majority of wireframes we record will use these defaults.

Making it required will only bloat the payloads in most cases where the wireframe uses default values. By making it optional we will safe 2 attributes per WF in majority of records.

Reply:
Well the thing is that I don't see the reason to provide defaults for them in that case. If they are optionals, their default is `null`.

Reply:
> Well the thing is that I don't see the reason to provide defaults for them in that case. If they are optionals, their default is null.

I don't mean providing defaults in JSON schema. I mean about **implementing** default in the player. // cc @DataDog/rum-session-replay 

I think we mix optionality with nullability here, but these are two separate concepts. By including `opacity` as `required` parameter it will be generated as:
```swift
let opacity: Double
```
on our side. No option to set `null` (Kotlin) or `nil` (Swift).

If we don't list `opacity` in `required`, it will be optional:
```swift
let opacity: Double?
```
hence we will be able to pass `nil` making the field absent in the JSON or (in Swift) `Optional<Double>.none` making it use `null` in the JSON.


Ref.: [Required Properties](https://json-schema.org/understanding-json-schema/reference/object.html#required-properties) on json-schema.org.

Reply:
Ok, that's clear now. We might go to provide a "default" in the JSON schema only for visibility but I am not sure. @ThibautGeriz what do you think, how do you prefer to have the defaults specified for each optional property ? Should  this be added in the schema also ?

Reply:
👍 alternatively we could guide player's implementation by explaining each default in the description.

Reply:
I think it's fine to have default in make the property optional. It's even better if we can match the browser default. For example for opacity, it's no opacity